assets/js/securesubmit.js

```diff
@@ -1,103 +1,378 @@
+
+                    var hps = new Heartland.HPS({
+                        publicKey: SecureSubmitObj.apiKey,
+                        cardNumber: _cardNumber,
+                        cardCvv: _cardCvv,
+                        cardExpMonth:_cardExpMonth,
+                        cardExpYear: _cardExpYear,
+                        success: function (response) {
+                            SecureSubmitObj.secureSubmitResponseHandler(response);
+                        },
+                        error: function (response) {
+                            SecureSubmitObj.secureSubmitResponseHandler(response);
+                        }
+                    });
+
+                    hps.tokenize();
+
+                    return false;
+                }
+
+                return true;
+            });
+        };
+
+        this.secureSubmitResponseHandler = function (response) {
+
+            var $form = this.form;
+            var ccInputPrefix = 'input_' + this.formId + '_' + this.ccFieldId + '_';
+            var ccInputSuffixes = ['1', '2_month', '2_year', '3'];
+            var i, input;
+
+            $('#securesubmit_response').remove();
+
+            if (!response.message) {
+                for (i = 0; i < ccInputSuffixes.length; i++) {
+                    input = $form.find('#' + ccInputPrefix + ccInputSuffixes[i]);
+                    input.val('');
+                }
+            }
+
+            $form.append($('<input type="hidden" name="securesubmit_response" id="securesubmit_response" />').val($.toJSON(response)));
+            $form.submit();
+        };
+
+        this.isCreditCardOnPage = function () {
+            var currentPage = this.getCurrentPageNumber();
+
+            // if current page is false or no credit card page number, assume this is not a multi-page form
+            if (!this.ccPage || !currentPage) {
+                return true;
+            }
+
+            return this.ccPage === currentPage;
+        };
+
+        this.getCurrentPageNumber = function () {
+            var currentPageInput = $('#gform_source_page_number_' + this.formId);
+            return currentPageInput.length > 0 ? parseInt(currentPageInput.val(), 10) : false;
+        };
+
+        this.init();
     };
+})(window, window.jQuery);
+
 
-    this.secureSubmitResponseHandler = function (response) {
-      var $form = this.form;
-      var ccInputPrefix = 'input_' + this.formId + '_' + this.ccFieldId + '_';
-      var ccInputSuffixes = ['1', '2_month', '2_year', '3'];
-      var i, input;
 
-      $('#securesubmit_response').remove();
 
-      if (!response.message) {
-        for (i = 0; i < ccInputSuffixes.length; i++) {
-          input = $form.find('#' + ccInputPrefix + ccInputSuffixes[i]);
-          input.val('');
+
+
+/*global $, jQuery*/
+var hps = (function ($) {
```

Comment:
The Secure Submit jQuery library wasn't a part of this integration before as it depended on the gateway hosted, non-jQuery library. This should be removed.